exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -146,33 +145,38 @@ func (b *Bitfinex) wsHandleData(respRaw []byte) error {
 			if !ok {
 				return errors.New("unable to type assert channel")
 			}
-			if symbol, ok := d["symbol"].(string); ok {
-				b.WsAddSubscriptionChannel(int(chanID),
-					channel,
-					symbol,
-				)
-			} else if key, ok := d["key"].(string); ok {
-				// Capture trading subscriptions
-				if contents := strings.Split(key, ":"); len(contents) > 3 {
-					// Edge case to parse margin strings.
-					// map[chanId:139136 channel:candles event:subscribed key:trade:1m:tXAUTF0:USTF0]
-					if contents[2][0] == 't' {
-						key = contents[2] + ":" + contents[3]
-					}
+			symbol, ok := d["symbol"].(string)
+			if !ok {
+				key, ok := d["key"].(string)
+				if !ok {
+					return fmt.Errorf("subscribed to channel but no symbol or key: %v", channel)
+				}
+				if channel != wsCandles {
+					// status channel not implemented at all yet.
+					return errors.Wrapf(common.ErrNotYetImplemented, "%v channel subscription keys", channel)
```

Comment:
Agreed. pkg/errors has been my default since [this go blog post](https://go.dev/blog/go1.13-errors) and I hadn't really considered before if it actually adds anything. It doesn't seem to.